test/service.go

```diff
@@ -35,17 +37,97 @@ func createPatch(cur, desired interface{}) ([]byte, error) {
 	return patch.MarshalJSON()
 }
 
-// CreateLatestService creates a service in namespace with the name names.Service
-// that uses the image specified by imagePath
-func CreateLatestService(logger *logging.BaseLogger, clients *Clients, names ResourceNames, imagePath string) (*v1alpha1.Service, error) {
-	service := LatestService(ServingNamespace, names, imagePath)
+// CreateRunLatestServiceReady creates a new RunLatest Service in state 'Ready'. This function expects Service and Image name passed in through 'names'.
+// Names is updated with the Route and Configuration created by the Service and ResourceObjects is returned with the Service, Route, and Configuration objects.
+// Returns error if the service does not come up correctly.
+func CreateRunLatestServiceReady(logger *logging.BaseLogger, clients *Clients, names *ResourceNames) (ResourceObjects, error) {
```

Comment:
This is doing a whole lot, I'd probably split it up into three functions

Reply:
This was attempting to pull out the Create work that all of our tests were doing, but doing slightly different. I do like the single function to create as it makes for an easy starting point for new tests, but I will look to see if I can refactor logic within the function out to new functions.